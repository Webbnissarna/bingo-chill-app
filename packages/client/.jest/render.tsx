import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
