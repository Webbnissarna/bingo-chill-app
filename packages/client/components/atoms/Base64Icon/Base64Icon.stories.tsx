import type { StoryDefault } from "@ladle/react";
import Base64Icon from ".";
import { SPOOKY_GHOST_IMAGE_BASE64 } from "@/.ladle/constants";

export function Default() {
  return (
    <Base64Icon description="Spooky ghost" src={SPOOKY_GHOST_IMAGE_BASE64} />
  );
}

export default {
  title: "Atoms / Base64 Icon",
} satisfies StoryDefault;
