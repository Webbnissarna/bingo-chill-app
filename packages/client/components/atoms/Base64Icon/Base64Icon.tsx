interface Base64IconProps {
  src: string;
  description?: string;
}

export default function Base64Icon({
  src,
  description,
}: Base64IconProps): JSX.Element {
  return <img src={src} alt={description ?? ""} />;
}
