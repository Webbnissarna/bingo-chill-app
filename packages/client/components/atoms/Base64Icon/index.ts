export { default } from "./Base64Icon";
