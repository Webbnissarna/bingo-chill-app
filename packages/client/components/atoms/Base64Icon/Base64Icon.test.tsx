import { render } from "../../../.jest/render";
import Base64Icon from ".";

describe("Base64Icon", () => {
  it("renders", () => {
    const { asFragment } = render(<Base64Icon src="" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
