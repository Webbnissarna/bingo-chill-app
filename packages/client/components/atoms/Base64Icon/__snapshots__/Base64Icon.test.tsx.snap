// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base64Icon renders 1`] = `
<DocumentFragment>
  <img
    alt=""
    class="w-full h-full object-contain"
    src=""
  />
</DocumentFragment>
`;
